libs/transactions/approval.ts

```diff
@@ -0,0 +1,47 @@
+import type Web3 from "web3";
+import path from "path";
+import { parentLogger } from "../utils";
+import type { Token } from "../tokens";
+import { submitTransaction } from "./helpers/submit";
+import type { TransactionReceipt } from "web3-core";
+
+const logger = parentLogger.child({
+  name: path.relative(process.cwd(), __filename),
+});
+
+export async function createAndSubmitApproval(
+  token: Token,
+  ownerAddress: string,
+  ownerPrivateKey: string,
+  spenderAddress: string,
+  value: string,
+  fee: string,
+  web3: Web3,
+): Promise<void | null | TransactionReceipt> {
+  logger.debug("createAndSubmitApproval");
+
+  const unsignedTx = await token.approveTransaction(
+    ownerAddress,
+    spenderAddress,
+    value,
+  );
+  // unsignedTx `null` signals that the approval is not required
```

Comment:
You meant `the submit is not required`?

Reply:
There's no need to submit because no unsigned transaction was returned since approval is not needed :see_no_evil: have updated the comment, perhaps more comprehensive now (?) [a53cdbb](https://github.com/maticnetwork/nightfall-sdk/pull/47/commits/a53cdbb0cd6ba5ced146bb60096db514c2858d15)